.github/workflows/ci-aarchxx-cross.yml

```
@@ -369,26 +273,18 @@ jobs:
-      if: failure() && github.ref == 'refs/heads/master'
-      uses: dawidd6/action-send-mail@v2
+  send-failure-notification:
+      uses: ./.github/workflows/failure-notification.yml
+      needs: [aarch64-cross-compile, arm-cross-compile, android-arm-cross-compile, android-aarch64-cross-compile, a64-on-x86]
+      # We need this to ensure this job runs even if there is failure in
+      # any of the workflows marked as needed.
+      if: ${{ always() }}
       with:
-        server_address: smtp.gmail.com
-        server_port: 465
-        username: ${{secrets.DYNAMORIO_NOTIFICATION_EMAIL_USERNAME}}
-        password: ${{secrets.DYNAMORIO_NOTIFICATION_EMAIL_PASSWORD}}
-        subject: |
-          [${{github.repository}}] ${{github.workflow}} FAILED
-          on ${{github.event_name}} at ${{github.ref}}
-        body: |
-          Github Actions CI workflow run FAILED!
-          Workflow: ${{github.workflow}}/a64-on-x86
-          Repository: ${{github.repository}}
-          Branch ref: ${{github.ref}}
-          SHA: ${{github.sha}}
-          Triggering actor: ${{github.actor}}
-          Triggering event: ${{github.event_name}}
-          Run Id: ${{github.run_id}}
-          See more details on github.com/DynamoRIO/dynamorio/actions/runs/${{github.run_id}}
-        to: dynamorio-devs@googlegroups.com
-        from: Github Action CI
+        test_suite_status: ${{ format('{0} {1} | {2} {3} | {4} {5} | {6} {7} | {8} {9}',
+                                      'aarch64-cross-compile', needs.aarch64-cross-compile.result,
+                                      'arm-cross-compile', needs.arm-cross-compile.result,
+                                      'android-arm-cross-compile', needs.android-arm-cross-compile.result,
+                                      'android-aarch64-cross-compile', needs.android-aarch64-cross-compile.result,
+                                      'a64-on-x86', needs.a64-on-x86.result) }}
```

Should we have comments by the job names to remember to update this if we change them, or add a new job?  If we change a name: would we get an error or silent failure?  Top-level comment saying to remember to update this job if any other job's name changes or one is added or removed?  Applies to each file.

---

>  If we change a name: would we get an error or silent failure? 

An error that says "Job 'A' depends on unknown job 'B'"

> Should we have comments by the job names to remember to update this if we change them, or add a new job?

Yes, I'll add top-level comments.

Also FWIW: See `test_suite_results_only: ${{ join(needs.*.result, ',') }}`: we pass all results using the `*` expression, so if we forget to add the status here for a new job and it fails on a master merge, the failure email would still be sent, and presumably someone would notice a failure email that doesn't include any failing job name.

I tried a bunch to somehow automate generation of status message also, but I couldn't find a way because the needs context disappointingly doesn't include the job name (https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#needs-context).

---

> we pass all results using the * expression, so...

What I said is true only if one forgets to add the new job to the status message, but still remembers to add it to `needs`.  Sorry for the confusion -- I did add the suggested top-level comment to remind future authors to add to the send-notification-email job.